src/opf/ptdf.jl

```diff
@@ -0,0 +1,112 @@
+using SparseArrays
+using PowerModels
+using LinearAlgebra
+
+abstract type AbstractPTDF end
+
+
+struct LazyPTDF{TF} <: AbstractPTDF
+    N::Int  # number of buses
+    E::Int  # number of branches
+    islack::Int  # Index of slack bus
+
+    A::SparseMatrixCSC{Float64,Int}  # incidence matrix
+    b::Vector{Float64}  # branch susceptances
+    BA::SparseMatrixCSC{Float64,Int}  # B*A
+    AtBA::SparseMatrixCSC{Float64,Int}  # AᵀBA
+
+    F::LDLt # Factorization of AᵀBA. Must be able to solve linear systems with F \ p
+
+    # TODO: cache
+end
+
+function LazyPTDF(data::OPFData)
+    N, E, A, b, ref_idx = data.N, data.E, data.A, data.b, data.ref_bus
+
+    all(data.branch_status .== 1) || error("LazyPTDF does not support disabled branches.")
```

Comment:
```suggestion
    all(data.branch_status) || error("LazyPTDF does not support disabled branches.")
```
Since `data.branch_status` is a `Vector{Bool}`